server/app/models/stack.rb

```diff
@@ -11,6 +11,8 @@ class Stack
   has_many :stack_revisions, dependent: :destroy
   has_many :stack_deploys, dependent: :destroy
   has_many :grid_services
+  has_many :volumes
+  embeds_many :external_volumes
```

Comment:
To make it more flexible and robust to define external volumes in stacks. They are kind-of mappings to grid level volumes where there might be extra attributes.